apps/ble_rpa_tool.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+from bumble.helpers import generate_irk, gen_rpa_with_irk, verify_rpa_with_irk
+
+
+def addr_bytes_to_str(addr_bytes: bytes) -> str:
+    return (
+        f'{addr_bytes[5]:02x}:{addr_bytes[4]:02x}:{addr_bytes[3]:02x}'
+        + f':{addr_bytes[2]:02x}:{addr_bytes[1]:02x}:{addr_bytes[0]:02x}'
+    )
+
+
+def addr_str_to_bytes(addr_str: str) -> bytes:
+    hex_bytes = addr_str.split(':')
+    return (
+        bytes.fromhex(hex_bytes[5])
+        + bytes.fromhex(hex_bytes[4])
+        + bytes.fromhex(hex_bytes[3])
+        + bytes.fromhex(hex_bytes[2])
+        + bytes.fromhex(hex_bytes[1])
+        + bytes.fromhex(hex_bytes[0])
+    )
+
+
+@click.group()
+def cli():
+    '''
+    This is a tool for generarting IRK, RPA,
+    and verifying IRK/RPA pairs
+    '''
+
+
+@click.command()
+def gen_irk() -> None:
+    print(generate_irk().hex())
+
+
+@click.command()
+@click.argument("irk", type=str)
+def gen_rpa(irk: str) -> None:
+    irk_bytes = bytes.fromhex(irk)
+    rpa = gen_rpa_with_irk(irk_bytes)
+    print(addr_bytes_to_str(rpa))
```

Comment:
You can use the `hci.Address` class for this: `hci.Address(rpa).to_string(with_type_qualifier=False)`.

Reply:
Done